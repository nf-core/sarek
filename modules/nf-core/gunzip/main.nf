process GUNZIP {
    tag "${archive}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/52/52ccce28d2ab928ab862e25aae26314d69c8e38bd41ca9431c67ef05221348aa/data'
        : 'community.wave.seqera.io/library/coreutils_grep_gzip_lbzip2_pruned:838ba80435a629f8'}"

    input:
    tuple val(meta), path(archive)

    output:
    tuple val(meta), path("${gunzip}"), emit: gunzip
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def extension = (archive.toString() - '.gz').tokenize('.')[-1]
    def name = archive.toString() - '.gz' - ".${extension}"
    def prefix = task.ext.prefix ?: name
    gunzip = prefix + ".${extension}"
    """
    # Not calling gunzip itself because it creates files
    # with the original group ownership rather than the
    # default one for that user / the work directory
    gzip \\
        -cd \\
        ${args} \\
        ${archive} \\
        > ${gunzip}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        gunzip: \$(echo \$(gunzip --version 2>&1) | sed 's/^.*(gzip) //; s/ Copyright.*\$//')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def extension = (archive.toString() - '.gz').tokenize('.')[-1]
    def name = archive.toString() - '.gz' - ".${extension}"
    def prefix = task.ext.prefix ?: name
    gunzip = prefix + ".${extension}"
    """
    touch ${gunzip}
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        gunzip: \$(echo \$(gunzip --version 2>&1) | sed 's/^.*(gzip) //; s/ Copyright.*\$//')
    END_VERSIONS
    """
}
