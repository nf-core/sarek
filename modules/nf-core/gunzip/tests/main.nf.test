nextflow_process {

    name "Test Process GUNZIP"
    script "../main.nf"
    process "GUNZIP"
    tag "gunzip"
    tag "modules_nfcore"
    tag "modules"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = Channel.of([
                        [],
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assertAll(
            { assert process.success },
            { assert snapshot(process.out).match() }
            )
        }

    }

    test("Should run without failures - prefix") {

        config './nextflow.config'

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = Channel.of([
                        [ id: 'test' ],
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assertAll(
            { assert process.success },
            { assert snapshot(process.out).match() }
            )
        }

    }

    test("Should run without failures - stub") {

        options '-stub'

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = Channel.of([
                        [],
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assertAll(
            { assert process.success },
            { assert snapshot(process.out).match() }
            )
        }

    }

    test("Should run without failures - prefix - stub") {

        options '-stub'
        config './nextflow.config'

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = Channel.of([
                        [ id: 'test' ],
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assertAll(
            { assert process.success },
            { assert snapshot(process.out).match() }
            )
        }

    }

}
