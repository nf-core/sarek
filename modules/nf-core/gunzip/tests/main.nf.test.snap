{
    "Should run without failures - prefix - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.xyz.fastq:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d327e4a19a6d5c5e974136cef8999d8c"
                ],
                "gunzip": [
                    [
                        {
                            "id": "test"
                        },
                        "test.xyz.fastq:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d327e4a19a6d5c5e974136cef8999d8c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-13T11:48:22.080222697"
    },
    "Should run without failures - stub": {
        "content": [
            {
                "0": [
                    [
                        [
                            
                        ],
                        "test_1.fastq:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d327e4a19a6d5c5e974136cef8999d8c"
                ],
                "gunzip": [
                    [
                        [
                            
                        ],
                        "test_1.fastq:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d327e4a19a6d5c5e974136cef8999d8c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-13T11:48:14.593020264"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        [
                            
                        ],
                        "test_1.fastq:md5,4161df271f9bfcd25d5845a1e220dbec"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d327e4a19a6d5c5e974136cef8999d8c"
                ],
                "gunzip": [
                    [
                        [
                            
                        ],
                        "test_1.fastq:md5,4161df271f9bfcd25d5845a1e220dbec"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d327e4a19a6d5c5e974136cef8999d8c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-13T11:48:01.295397925"
    },
    "Should run without failures - prefix": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.xyz.fastq:md5,4161df271f9bfcd25d5845a1e220dbec"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d327e4a19a6d5c5e974136cef8999d8c"
                ],
                "gunzip": [
                    [
                        {
                            "id": "test"
                        },
                        "test.xyz.fastq:md5,4161df271f9bfcd25d5845a1e220dbec"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d327e4a19a6d5c5e974136cef8999d8c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-13T11:48:07.414271387"
    }
}